tests/Application.Tests.Integration/Folders/Commands/AddFolderTests.cs

```diff
@@ -0,0 +1,251 @@
+    {
+    }
+
+    [Fact]
+    public async Task ShouldAddFolder_WhenAddDetailsAreValid()
+    {
+        // Arrange
+        var addRoomCommand = _roomGenerator.Generate();
+        var room = await SendAsync(addRoomCommand);
+
+        var addLockerCommand = _lockerGenerator.Generate();
+        addLockerCommand = addLockerCommand with
+        {
+            RoomId = room.Id,
+            Capacity = 1
+        };
+        var locker = await SendAsync(addLockerCommand);
+        
+        var addFolderCommand = _folderGenerator.Generate();
+        addFolderCommand =  addFolderCommand with 
+        {
+            LockerId = locker.Id,
+            Capacity = 1
+        };
+
+        // Act
+        var folder = await SendAsync(addFolderCommand);
+        // Assert
+        folder.Name.Should().Be(addFolderCommand.Name);
+        folder.Description.Should().Be(addFolderCommand.Description);
+        folder.Capacity.Should().Be(addFolderCommand.Capacity);
+        folder.Locker.Id.Should().Be(locker.Id);
+        folder.Locker.NumberOfFolders.Should().Be(locker.NumberOfFolders + 1);
+        folder.NumberOfDocuments.Should().Be(0);
+        folder.IsAvailable.Should().BeTrue();
+        
+        // Clean up
+        var folderEntity = await FindAsync<Folder>(folder.Id);
+        var lockerEntity = await FindAsync<Locker>(locker.Id);
+        var roomEntity = await FindAsync<Room>(room.Id);
+        Remove(folderEntity);
+        Remove(lockerEntity);
+        Remove(roomEntity);
+    }
+
+    [Fact]
+    public async Task ShouldAddFolder_WhenFoldersHasSameNameButInDifferentLockers()
+    {
+        // Arrange
+        var sameFolderName = new Faker().Commerce.ProductName();
+        var addRoomCommand = _roomGenerator.Generate();
+        addRoomCommand = addRoomCommand with
+        {
+            Capacity = 2
+        };
+        
+        var room = await SendAsync(addRoomCommand);
+        
+        var addLockerCommand = _lockerGenerator.Generate();
+        
+        var addLockerACommand = addLockerCommand with
+        {
+            RoomId = room.Id,
+            Capacity = 1
+        };
+        var addLockerBCommand = addLockerCommand with
+        {
+            RoomId = room.Id,
+            Name = new Faker().Commerce.ProductName(),
+            Capacity = 1
+        };
+        
+        var lockerA = await SendAsync(addLockerACommand);
+        var lockerB = await SendAsync(addLockerBCommand);
+        
+        var addFolderCommand = _folderGenerator.Generate();
+        var addFolderCommandForLockerA =  addFolderCommand with 
+        {
+            Name = sameFolderName,
+            LockerId = lockerA.Id
+        };
+        var addFolderCommandForLockerB =  addFolderCommand with 
+        {
+            Name = sameFolderName,
+            LockerId = lockerB.Id
+        };
+        // Act
+        var folderA = await SendAsync(addFolderCommandForLockerA);
```

Comment:
This should not be in Act

Reply:
fixed at https://github.com/ProFile-org/ProFile-BE/pull/80/commits/001ea4b4f149a2b55bb60565c921ca2c54419550